cmd/skopeo/utils.go

```diff
@@ -180,12 +184,23 @@ func (opts *imageOptions) newSystemContext() (*types.SystemContext, error) {
 	if opts.credsOption.present && opts.noCreds {
 		return nil, errors.New("creds and no-creds cannot be specified at the same time")
 	}
+	if opts.credsOption.present && opts.userName.present {
+		return nil, errors.New("creds and username cannot be specified at the same time")
+	}
+	if opts.password.present && !opts.userName.present {
+		return nil, errors.New("password cannot be specified without a username")
+	}
 	if opts.credsOption.present {
 		var err error
 		ctx.DockerAuthConfig, err = getDockerAuth(opts.credsOption.value)
 		if err != nil {
 			return nil, err
 		}
+	} else if opts.userName.present {
+		ctx.DockerAuthConfig = &types.DockerAuthConfig{
+			Username: opts.userName.value,
+			Password: opts.password.value,
```

Comment:
… strictly speaking `opts.password.value` is not valid to use `if !opts.password.present`; so if the case of `--username` without `--password` should be accepted, this should probably add an explicit condition.

It just seems simpler to me to require `opts.userName.present == opts.password.present`, and if someone really has an account with an empty password, force them to use `--password ''` explicitly. That is simpler for us to implement, and even for the users it generalizes better.